util/new_keyboard.sh

```diff
@@ -147,6 +147,9 @@ username=$prompt_return
 
 echo
 
+# Create new keyboard directory and any intermediate directories.
+mkdir -p "$keyboard_dir"
```

Comment:
I would rather this be at the beginning of `copy_templates`.

Reply:
That's where I had put it initially, but then I saw that all of these functions that are called below kind of use `$keyboard_dir` “equally” — there wasn't anything in `copy_templates()` in particular that screamed “_this_ function should be where the keyboard directories are created if they don't exist”. I hope that makes sense.

Reply:
But I'll have to rework this either way since the rest of the script isn't working, so I may end up putting it into one of the functions, or moving it to a new function entirely.

Reply:
The `replace_*_placeholders()` functions only assume the files they're S&Ring exist, which `copy_templates()` takes care of - this in turn means it needs to create the keyboard directory if *it* doesn't exist. At the moment the first `cp` does this implicitly, which of course is not correct.